# INFERENTIAL STATISTICS

## Lesson 10a: t-Tests, Part 1

z-test only works when we know population mean and standard deviation

#### First example

In [1]:
import numpy as np
import pandas as pd

finches_data=pd.read_csv('data/finches.csv', sep=',')

sample_size = len(finches_data)
print sample_size

500


In [2]:
sample_mean = np.mean(finches_data.values)
print sample_mean

6.4696


In [3]:
# ddof used when we're calculating samples so we devide with n-1 instead of just n
sample_deviation = np.std(finches_data.values, ddof=1)
print sample_deviation

0.396118239063


In [4]:
population_mean = 6.07

t = (sample_mean - population_mean) / (sample_deviation / (sample_size ** 0.5) )
print t

22.5572234675


#### Second example

In [5]:
sample = [5,19,11,23,12,7,3,21]
second_population_mean = 10
second_sample_mean = np.mean(sample)
sample_std = np.std(sample, ddof=1)
no_samples = len(sample)
print no_samples

t = (second_sample_mean - second_population_mean) / (sample_std / (no_samples ** 0.5) )
print t

8
0.977461894334


#### Third example

Calculating p value from t statistic

from scipy.stats import t

nop_tails_test = 2
print t.sf(0.977, 7) * nop_tails_test

#### Fourth example

Santa Clara Appartments

In [6]:
house_mean = 1830
sample_no = 25
sample_mean = 1700
sample_std = 200

t = (sample_mean - house_mean) / (sample_std / (sample_no ** 0.5) )
print t

-3.25


In [7]:
cohens_d = float(sample_mean - house_mean) / (sample_std)
print cohens_d

-0.65


#### Dependent t-tests for paired samples

In [8]:
keyboard_data=pd.read_csv('data/keyboard.csv', sep=',')

print keyboard_data

sample_difference = keyboard_data["QWERTY errors"].values - keyboard_data["Alphabetical errors"].values

difference_std = np.std(sample_difference, ddof=1)
print difference_std

    QWERTY errors  Alphabetical errors
0               6                    6
1               6                   11
2               2                    8
3               7                    5
4               8                   11
5               8                    8
6               2                   10
7               3                    7
8               5                    4
9               7                    3
10             10                    7
11              5                    6
12              4                   10
13              7                   10
14              5                    6
15              7                    5
16              4                   10
17              5                   11
18              2                   13
19              5                    8
20              3                    5
21              4                   11
22              4                    7
23              4                    8
24              4        

**==> Calculating confidence interval**

In [9]:
referent_mean = np.mean(keyboard_data["QWERTY errors"].values)
new_mean = np.mean(keyboard_data["Alphabetical errors"].values)
t_critical = 2.064 # t value for alpha value of 0.05

confidence_interval_start = (referent_mean - new_mean) - t_critical * difference_std / (sample_no ** 0.5)
confidence_interval_end = (referent_mean - new_mean) + t_critical * difference_std / (sample_no ** 0.5)

print "( " + str(confidence_interval_start) + ", " + str(confidence_interval_end) + " )"

( -4.24382293145, -1.19617706855 )


We can conclude that on average users will make between 4.24 and 1.2 less errors on QWERTY than on alphabetical keyboard